### PR TITLE
Include specifications for the WorkflowHub in the docs

### PR DESCRIPTION
**Pull request by AngryMaciek**
As previously discussed with @johanneskoester, I included specifications for a *snakemake* repository structure that would comply with WorkflowHub.

### CODE REVIEW

File: docs/snakefiles/deployment.rst [hunk 1]
```diff
@@ -67,6 +67,61 @@ Given that cookiecutter is installed, you can use it via:
...
+
+To adhere to the high WorkflowHub standards of scientific workflows the recommended *snakemake* repository structure presented above needs to be extended by the following elements:
+
+- Code of Conduct
+- Contribution instructions
+- Workflow rule graph
+- Workflow documentation
+- Test directory
+
+A code of conduct for the repository developers as well as instruction on how to contribute to the project should be placed in the top-level files: ``CODE_OF_CONDUCT.md`` and ``CONTRIBUTING.md``, respectively. Each *snakemake* workflow repository needs to contain an SVG-formatted rule graph placed in a subdirectory ``images/rulegraph.svg``. Additionally, the workflow should be annotated with a technical documentation of all of its subsequent steps, described in ``workflow/documentation.md``. Finally, the repository should contain a ``.tests`` directory with two subdirectories: ``.tests/integration`` and ``.tests/unit``. The former has to contain all the input data, configuration specifications and shell commands required to run an integration test of the whole workflow. The latter shall contain subdirectories dedicated to testing each of the separate workflow steps independently.
```
Comment (johanneskoester) [inline docs/snakefiles/deployment.rst:84]: For the former, please link to `snakefiles_testing_` (see [here](https://snakemake.readthedocs.io/en/latest/snakefiles/testing.html)) and mention that Snakemake is able to automatically generate the unit tests.

**Review by johanneskoester — CHANGES_REQUESTED** [review]
Awesome, thanks a lot! Just one thing.

File: docs/snakefiles/deployment.rst (hunk 1 above)
Comment (AngryMaciek) [reply docs/snakefiles/deployment.rst]: That's a really cool feature I was not aware of!
OK, Done in c5c37e6.

Comment (johanneskoester) [reply docs/snakefiles/deployment.rst]: Well, I implemented it when I saw your PR. Thought it might help a lot here :-).

**Pull request merged**